src/PollingBlockTracker.ts

```diff
@@ -55,17 +85,166 @@ export class PollingBlockTracker extends BaseBlockTracker {
+    this.on('removeListener', this._onRemoveListener);
+  }
+
+  private _onNewListener(eventName: string | symbol): void {
+    // `newListener` is called *before* the listener is added
+    if (blockTrackerEvents.includes(eventName)) {
+      // TODO: Handle dangling promise
+      this._maybeStart();
+    }
+  }
+
+  private _onRemoveListener(): void {
+    // `removeListener` is called *after* the listener is removed
+    if (this._getBlockTrackerEventCount() > 0) {
+      return;
+    }
+    this._maybeEnd();
+  }
+
+  private async _maybeStart(): Promise<void> {
+    if (this._isRunning) {
+      return;
+    }
+    this._isRunning = true;
+    // cancel setting latest block to stale
+    this._cancelBlockResetTimeout();
+    await this._start();
+    this.emit('_started');
+  }
+
+  private async _maybeEnd(): Promise<void> {
+    if (!this._isRunning) {
+      return;
+    }
+    this._isRunning = false;
+    this._setupBlockResetTimeout();
+    await this._end();
+    this.emit('_ended');
+  }
+
+  private _getBlockTrackerEventCount(): number {
+    return blockTrackerEvents
+      .map((eventName) => this.listenerCount(eventName))
+      .reduce(calculateSum);
+  }
+
+  private _shouldUseNewBlock(newBlock: string) {
+    const currentBlock = this._currentBlock;
+    if (!currentBlock) {
+      return true;
+    }
+    const newBlockInt = hexToInt(newBlock);
+    const currentBlockInt = hexToInt(currentBlock);
+
+    return (
+      (this._usePastBlocks && newBlockInt < currentBlockInt) ||
+      newBlockInt > currentBlockInt
+    );
+  }
+
+  private _newPotentialLatest(newBlock: string): void {
+    if (!this._shouldUseNewBlock(newBlock)) {
+      return;
+    }
+    this._setCurrentBlock(newBlock);
+  }
+
+  private _setCurrentBlock(newBlock: string): void {
+    const oldBlock = this._currentBlock;
+    this._currentBlock = newBlock;
+    this.emit('latest', newBlock);
+    this.emit('sync', { oldBlock, newBlock });
+  }
+
+  private _setupBlockResetTimeout(): void {
+    // clear any existing timeout
+    this._cancelBlockResetTimeout();
+    // clear latest block when stale
+    this._blockResetTimeout = setTimeout(
+      this._resetCurrentBlock,
+      this._blockResetDuration,
+    );
+
+    // nodejs - dont hold process open
+    if (this._blockResetTimeout.unref) {
+      this._blockResetTimeout.unref();
+    }
+  }
+
+  private _cancelBlockResetTimeout(): void {
+    if (this._blockResetTimeout) {
+      clearTimeout(this._blockResetTimeout);
+    }
+  }
+
+  private _resetCurrentBlock(): void {
+    this._currentBlock = null;
+  }
+
   // trigger block polling
   async checkForLatestBlock() {
     await this._updateLatestBlock();
     return await this.getLatestBlock();
   }
 
-  protected async _start(): Promise<void> {
+  private async _start(): Promise<void> {
     this._synchronize();
   }
 
-  protected async _end(): Promise<void> {
+  private async _end(): Promise<void> {
```

Comment:
Do we need this method?

Reply:
I don't think so! I tried to limit the scope of this PR though. I started going down the path of simplifying things, but ended up going too far so I scrapped it.